### PR TITLE
Starknet Tx Execution

### PR DESCRIPTION
**Pull request by EvolveArt**
# Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Testing
- [ ] Other (please describe):

# What is the current behavior?

Issue Number: #21 

# What is the new behavior?

Adds logic to submit `Invoke` transactions to the sequencer.
- Check if address exists within the pallet's storage
- Validate transaction (runs `__validate__ entrypoint)
- Execute transaction (runs `__execute__` entrypoint)
- Add to `pending` list

### CODE REVIEW

File: crates/pallets/starknet/src/lib.rs [hunk 1]
```diff
@@ -143,19 +153,47 @@ pub mod pallet {
...
+
+    /// Mapping from Starknet contract storage key to its value.
+    #[pallet::storage]
+    #[pallet::getter(fn storage)]
+    pub(super) type StorageView<T: Config> = StorageMap<_, Twox64Concat, ContractStorageKey, StarkFelt, ValueQuery>;
+
+    /// Mapping from Starknet class hash to class
+    // #[pallet::storage]
+    // #[pallet::getter(fn class)]
+    // pub(super) type ContractClasses<T: Config> = StorageMap<_, Twox64Concat, ContractClassHash, ClassInfo,
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:169]: please remove commented code

File: crates/pallets/starknet/src/lib.rs [hunk 2]
```diff
@@ -261,7 +338,7 @@ pub mod pallet {
             let pending = Self::pending();
 
             let global_state_root = U256::zero();
-            let sequencer_address = U256::zero();
+            let sequencer_address = [0; 32];
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:341]: can we use the custom type you defined instead please

File: crates/pallets/starknet/src/lib.rs [hunk 3]
```diff
@@ -295,17 +372,97 @@ pub mod pallet {
...
+                .collect();
+
+            // let class_hash_to_class: ContractClassMapping = ContractClasses::<T>::iter().map(|(key, value)| {
+            // 	let class_hash = ClassHash(StarknetStarkFelt::new(key).unwrap());
+            // 	let contract_class = ContractClass::try_from(value).unwrap();
+            // 	(class_hash, contract_class)
+            // }).collect();
+            // TODO: remove this when declare is implemented
+            let class_hash_to_class: ContractClassMapping = HashMap::from([
+                (
```
Comment (abdelhamidbakhta) [inline crates/pallets/starknet/src/lib.rs:436]: we need to think how to iterate with some mocked/partial features going forward, as well as proper testing

**Review by abdelhamidbakhta — COMMENTED** [review]
left some comments waiting for CI to pass before doing a more complete review

File: crates/pallets/starknet/src/lib.rs (hunk 3 above)
Comment (EvolveArt) [reply crates/pallets/starknet/src/lib.rs:436]: This is just temporary, as once we have declare transactions we can remove it.
Proper mocking/testing is done through the genesis config initialisation in `mock.rs` and `test.rs` like it's done for contract addresses and class hashes.

**Review by abdelhamidbakhta — APPROVED** [review]
lgtm

**Pull request merged**